src/sdam/server.ts

```diff
@@ -264,26 +288,37 @@ class Server extends EventEmitter {
       return;
     }
 
-    this.s.pool.withConnection((err: any, conn: any, cb: any) => {
-      if (err) {
+    this.s.pool.withConnection((err, conn, cb) => {
+      if (err || !conn) {
         markServerUnknown(this, err);
         return cb(err);
       }
 
-      conn.command(ns, cmd, options, makeOperationHandler(this, conn, cmd, options, cb));
+      conn.command(
+        ns,
+        cmd,
+        options,
+        makeOperationHandler(this, conn, cmd, options, cb) as Callback
+      );
     }, callback);
   }
 
   /**
    * Execute a query against the server
    *
-   * @param {string} ns The MongoDB fully qualified namespace (ex: db1.collection1)
-   * @param {object} cmd The command document for the query
-   * @param {any} cursorState
-   * @param {object} options Optional settings
-   * @param {Function} callback
+   * @param ns The MongoDB fully qualified namespace (ex: db1.collection1)
+   * @param cmd The command document for the query
+   * @param cursorState
+   * @param options Optional settings
+   * @param callback
    */
-  query(ns: string, cmd: object, cursorState: any, options: object, callback: Function) {
+  query(
+    ns: string,
+    cmd: Document,
+    cursorState: InternalCursorState,
+    options: QueryOptions,
+    callback: Callback
+  ) {
     if (this.s.state === STATE_CLOSING || this.s.state === STATE_CLOSED) {
       callback(new MongoError('server is closed'));
       return;
```

Comment:
A few lines down there's an untyped call: `withConnection((err: any, conn: any, cb: any) => {`, you can ctrl-f for this  there should be two other instances that can be typed.